### PR TITLE
Add semtype relations for array

### LINKED ISSUE #34629: Port SemType Tests from nBallerina to lang repo Tests
$Subject.
No of people: 2

### PR DESCRIPTION
**Pull request by chiranSachintha**
Fixes #34629 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by KavinduZoysa** [comment]
@chiranSachintha, shall we break the tests based on `// @type` comment. Each file can be included test cases related to one or two `// @type`.

File: tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java [hunk 1]
```diff
@@ -81,7 +82,7 @@ public void testSubTypeRelationship(String filePath) throws IOException {
         List<BTypeDefinitionSymbol> bTypeDefinitionSymbols = new ArrayList<>();
         for (Scope.ScopeEntry value : bLangPackage.symbol.scope.entries.values()) {
             BSymbol bSymbol = value.symbol;
-            if (bSymbol.kind == SymbolKind.TYPE_DEF) {
+            if (bSymbol.kind == SymbolKind.TYPE_DEF && bSymbol.origin == SymbolOrigin.SOURCE) {
```
Comment (pcnfernando) [inline tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java:85]: Shouldn't this be ! Virtual?

**Comment by KavinduZoysa** [comment]
@chiranSachintha did not you find any failing test case?

File: tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java (hunk 1 above)
Comment (chiranSachintha) [reply tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java:85]: We are going to test the relations among the type definitions defined in the source file.  So the origin should be `source` right? (It cannot be `BUILTIN`)

**Comment by chiranSachintha** [comment]
> @chiranSachintha did not you find any failing test cases?

No, found test cases that are not supported by the ballerina compiler(added them to -d.bal file)

File: tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java (hunk 1 above)
Comment (pcnfernando) [reply tests/jballerina-semtype-test/src/test/java/io/ballerina/test/SemTypeTest.java:85]: If it doesn't consider balas; it's fine.

**Review by KavinduZoysa — APPROVED** [review]

**Pull request merged**